prow/sidecar/run.go

```diff
@@ -121,13 +122,40 @@ func (o Options) Run() error {
 		aborted = returnCode == 130
 	}
 
-	return o.doUpload(spec, passed, aborted)
+	metadataFile := o.WrapperOptions.MetadataFile
+	if _, err := os.Stat(metadataFile); err != nil {
+		if !os.IsNotExist(err) {
+			logrus.WithError(err).Errorf("Failed to stat %s", metadataFile)
+		}
+		return o.doUpload(spec, passed, aborted, nil)
+	}
+
+	metadataRaw, err := ioutil.ReadFile(metadataFile)
+	if err != nil {
+		logrus.WithError(err).Errorf("cannot read %s", metadataFile)
+		return o.doUpload(spec, passed, aborted, nil)
+	}
+
+	metadata := map[string]string{}
+	if err := json.Unmarshal(metadataRaw, &metadata); err != nil {
+		logrus.WithError(err).Errorf("Failed to unmarshal %s", metadataFile)
+		return o.doUpload(spec, passed, aborted, nil)
+	}
+	return o.doUpload(spec, passed, aborted, metadata)
```

Comment:
Just defer this call on line 124 and you can avoid including it in all the error handling cases.